lib/bson.js

```diff
@@ -213,222 +214,7 @@ function deserializeStream(data, startIndex, numberOfDocuments, documents, docSt
-/**
- * Binary BSON Type
- *
- * @classconstant BSON_DATA_BINARY
- **/
-const BSON_DATA_BINARY = 5;
-
-/**
- * ObjectID BSON Type
- *
- * @classconstant BSON_DATA_OID
- **/
-const BSON_DATA_OID = 7;
-
-/**
- * Boolean BSON Type
- *
- * @classconstant BSON_DATA_BOOLEAN
- **/
-const BSON_DATA_BOOLEAN = 8;
-
-/**
- * Date BSON Type
- *
- * @classconstant BSON_DATA_DATE
- **/
-const BSON_DATA_DATE = 9;
-
-/**
- * null BSON Type
- *
- * @classconstant BSON_DATA_NULL
- **/
-const BSON_DATA_NULL = 10;
-
-/**
- * RegExp BSON Type
- *
- * @classconstant BSON_DATA_REGEXP
- **/
-const BSON_DATA_REGEXP = 11;
-
-/**
- * Code BSON Type
- *
- * @classconstant BSON_DATA_CODE
- **/
-const BSON_DATA_CODE = 13;
-
-/**
- * Symbol BSON Type
- *
- * @classconstant BSON_DATA_SYMBOL
- **/
-const BSON_DATA_SYMBOL = 14;
-
-/**
- * Code with Scope BSON Type
- *
- * @classconstant BSON_DATA_CODE_W_SCOPE
- **/
-const BSON_DATA_CODE_W_SCOPE = 15;
-
-/**
- * 32 bit Integer BSON Type
- *
- * @classconstant BSON_DATA_INT
- **/
-const BSON_DATA_INT = 16;
-
-/**
- * Timestamp BSON Type
- *
- * @classconstant BSON_DATA_TIMESTAMP
- **/
-const BSON_DATA_TIMESTAMP = 17;
-
-/**
- * Long BSON Type
- *
- * @classconstant BSON_DATA_LONG
- **/
-const BSON_DATA_LONG = 18;
-
-/**
- * MinKey BSON Type
- *
- * @classconstant BSON_DATA_MIN_KEY
- **/
-const BSON_DATA_MIN_KEY = 0xff;
-
-/**
- * MaxKey BSON Type
- *
- * @classconstant BSON_DATA_MAX_KEY
- **/
-const BSON_DATA_MAX_KEY = 0x7f;
-
-/**
- * Binary Default Type
- *
- * @classconstant BSON_BINARY_SUBTYPE_DEFAULT
- **/
-const BSON_BINARY_SUBTYPE_DEFAULT = 0;
-
-/**
- * Binary Function Type
- *
- * @classconstant BSON_BINARY_SUBTYPE_FUNCTION
- **/
-const BSON_BINARY_SUBTYPE_FUNCTION = 1;
-
-/**
- * Binary Byte Array Type
- *
- * @classconstant BSON_BINARY_SUBTYPE_BYTE_ARRAY
- **/
-const BSON_BINARY_SUBTYPE_BYTE_ARRAY = 2;
-
-/**
- * Binary UUID Type
- *
- * @classconstant BSON_BINARY_SUBTYPE_UUID
- **/
-const BSON_BINARY_SUBTYPE_UUID = 3;
-
-/**
- * Binary MD5 Type
- *
- * @classconstant BSON_BINARY_SUBTYPE_MD5
- **/
-const BSON_BINARY_SUBTYPE_MD5 = 4;
-
-/**
- * Binary User Defined Type
- *
- * @classconstant BSON_BINARY_SUBTYPE_USER_DEFINED
- **/
-const BSON_BINARY_SUBTYPE_USER_DEFINED = 128;
-
-module.exports = {
-  // constants
-  BSON_INT32_MAX,
-  BSON_INT32_MIN,
-  BSON_INT64_MAX,
-  BSON_INT64_MIN,
-  JS_INT_MAX,
-  JS_INT_MIN,
-  BSON_DATA_NUMBER,
-  BSON_DATA_STRING,
-  BSON_DATA_OBJECT,
-  BSON_DATA_ARRAY,
-  BSON_DATA_BINARY,
-  BSON_DATA_OID,
-  BSON_DATA_BOOLEAN,
-  BSON_DATA_DATE,
-  BSON_DATA_NULL,
-  BSON_DATA_REGEXP,
-  BSON_DATA_CODE,
-  BSON_DATA_SYMBOL,
-  BSON_DATA_CODE_W_SCOPE,
-  BSON_DATA_INT,
-  BSON_DATA_TIMESTAMP,
-  BSON_DATA_LONG,
-  BSON_DATA_MIN_KEY,
-  BSON_DATA_MAX_KEY,
-  BSON_BINARY_SUBTYPE_DEFAULT,
-  BSON_BINARY_SUBTYPE_FUNCTION,
-  BSON_BINARY_SUBTYPE_BYTE_ARRAY,
-  BSON_BINARY_SUBTYPE_UUID,
-  BSON_BINARY_SUBTYPE_MD5,
-  BSON_BINARY_SUBTYPE_USER_DEFINED,
-
+module.exports = Object.assign(constants, {
```

Comment:
This _may_ mess with bundlers trying to read the exported names.

Reply:
Yeah I thought maybe that would be the case, if we actually drop Node 4 support we can just use the spread operator here which might make it nicer

Reply:
This can be fixed with `Object.assign({}, constants, {...})`.